ansible_events/websocket.py

```diff
@@ -1,12 +1,60 @@
+import base64
 import json
 import logging
 from asyncio.exceptions import CancelledError
 
 import websockets
+import yaml
+
+from ansible_events import rules_parser as rules_parser
+from ansible_events.key import install_private_key
 
 logger = logging.getLogger("ansible_events.websocket")
 
 
+async def request_workload(activation_id, websocket_address):
+    logger.info("websocket %s connecting", websocket_address)
+    async for websocket in websockets.connect(websocket_address):
```

Comment:
This code looks odd. Can you please explain the logic here?

If that's a one time connection, you should use `async with websocket.connect(...) as websocket`.
If the intention is to automatically reconnect on disconnects, the `websockets.ConnectionClosed` exception should be handled.